src/components/Table.js

```diff
@@ -0,0 +1,52 @@
+import React, { useState, useEffect, useCallback } from "react"
+
+import { formatAmount, formatDate } from "../utils/formaters"
+
+const Table = ({ deals, selected }) => {
+  const [currentDeals, setCurrentDeals] = useState(deals)
+
+  const handleFilter = useCallback(() => {
+    if (selected) {
+      const filteredDeals = deals.filter((deal) => formatDate(deal.createdAt) === selected)
+      setCurrentDeals(filteredDeals)
+    } else {
+      setCurrentDeals(deals)
+    }
+  }, [selected, deals])
```

Comment:
Given that we had 100 deals on our database and we were paginating the deals 10 per page.
Would this solution always works? 
Yes/No - Why?